node/enable_libp2p.go

```diff
@@ -0,0 +1,40 @@
+package node
+
+import (
+	"github.com/filecoin-project/lotus/node/config"
+	"github.com/filecoin-project/lotus/node/repo"
+)
+
+func ApplyIfEnableLibP2P(r repo.Repo, opts ...Option) Option {
```

Comment:
This looks really hacky, I'd love to avoid having this if at all possible

Reply:
I have removed `ApplyIfEnableLibP2P` so no unnecessary abstraction now. Looking into passing config into `StorageMiner` to see if we can simplify further.

Reply:
Having passed the Subsystems config, and adding an `enableLibp2pNode bool` in `Settings`, this is no longer needed.